### ISSUE TITLE
Support omitting geometry to enable smaller payloads and faster retrieval

### ISSUE THREAD

**Issue by alpha-beta-soup** [geopython/pygeoapi#384]
**Is your feature request related to a problem? Please describe.**

Sometimes a client doesn't actually care about a geometry, but the properties alone. To reduce the payload in such a use-case, it could be useful to allow clients to express that they want to omit a geometry altogether.

As discussed in https://github.com/geopython/pygeoapi/issues/363, we might also like to have an alternative non-WGS84 geometry representation as a GeoJSON `property` (WKT string or other value), and then the ability to disable the `geometry` allows us to still keep payloads reasonably small, particularly for large and complex geometries.

**Describe the solution you'd like**

Some URL parameter would need to be chosen to allow a client to express this preference, probably a Boolean `geom=false` — this is a very loose suggestion and I eagerly await better ideas. The default behaviour should of course be to include the geometry.

The [GeoJSON spec](https://geojson.org/geojson-spec.html#feature-objects) says:

> A GeoJSON object with the type "Feature" is a feature object.
>
> A feature object must have a member with the name "geometry". The value of the geometry member is a geometry object as defined above or a JSON null value.

So the `geometry` property would still need to be included in any GeoJSON responses, but it would have the value of `null`.

An exploration of how this should be handled in (Geo)JSON-LD should be carried out—we do not want to make a _positive statement_ that a geometry *is* null or empty (unless it actually is), but rather that it is simply absent, not-represented, or undefined in this case.

Other output types would need to be considered too.

**Describe alternatives you've considered**

The alternative is to *not* support this and instead always include the geometry. The drawback is simply a larger payload, slower download, and also potentially wasted server time retrieving/joining/reprojecting geometries that will be ignored.

**Additional context**

Ideally this is implemented more deeply than simply lopping off the geometry at the last moment, but actually results in geometries being totally ignored when requests are made to underlying data stores, unless there's a spatial query (e.g. a bbox filter) being simultaneously applied.

**Comment by francbartoli** [comment]
@alpha-beta-soup the feature sounds reasonable. Regarding the implementation I'm wondering if it's just a matter of the requested `Content-Type`:

```text
/items
/items/{item_id}
application/geo+json --> return GeoJSON
application/json          --> return JSON encode properties 
```

What do you think?

**Comment by alpha-beta-soup** [comment]
Possibly, but it's difficult because we alllow the `f` parameter to override the `Content-Type`, and `f=json` currently means GeoJSON. This is partly why I made #317 so there isn't this ambiguity.

**Comment by dblodgett-usgs** [comment]
I would vote for optional or even default behavior for the json-ld media type that would omit the geometry for anything other than a point featuretype.

A semi-proper way to do this would be to have the `geojson:geometry` property be like in the example below: ([see in the playground here](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22%40context%22%3A%5B%22https%3A%2F%2Fgeojson.org%2Fgeojson-ld%2Fgeojson-context.jsonld%22%2C%7B%22schema%22%3A%22https%3A%2F%2Fschema.org%22%2C%22NAME%22%3A%22schema%3Aname%22%2C%22gnis_url%22%3A%22schema%3AsubjectOf%22%2C%22uri%22%3A%22%40id%22%7D%5D%2C%22type%22%3A%22Feature%22%2C%22properties%22%3A%7B%22uri%22%3A%22https%3A%2F%2Fgeoconnex.us%2Fref%2Fhu02%2F09%22%2C%22NAME%22%3A%22Souris-Red-Rainy%20Region%22%2C%22gnis_url%22%3A%22https%3A%2F%2Fgeonames.usgs.gov%2Fapex%2Ff%3Fp%3Dgnispq%3A3%3A%3A%3ANO%3A%3AP3_FID%3A2730139%22%2C%22GNIS_ID%22%3A2730139%2C%22HUC2%22%3A%2209%22%2C%22LOADDATE%22%3A%222019%2F05%2F23%2018%3A26%3A19%2B00%22%7D%2C%22geometry%22%3A%7B%22schema%3AencodingFormat%22%3A%22application%2Fgeo%2Bjson%22%2C%22schema%3Aurl%22%3A%22https%3A%2F%2Finfo.geoconnex.us%2Fcollections%2Fhu02%2Fitems%2F09%3Ff%3Djson%22%7D%2C%22id%22%3A%22https%3A%2F%2Fgeoconnex.us%2Fref%2Fhu02%2F09%22%2C%22links%22%3A%5B%7B%22rel%22%3A%22alternate%22%2C%22type%22%3A%22application%2Fgeo%2Bjson%22%2C%22title%22%3A%22This%20document%20as%20GeoJSON%22%2C%22href%22%3A%22https%3A%2F%2Finfo.geoconnex.us%2Fcollections%2Fhu02%2Fitems%2F09%3Ff%3Djson%22%7D%2C%7B%22rel%22%3A%22self%22%2C%22type%22%3A%22application%2Fld%2Bjson%22%2C%22title%22%3A%22This%20document%20as%20RDF%20(JSON-LD)%22%2C%22href%22%3A%22https%3A%2F%2Finfo.geoconnex.us%2Fcollections%2Fhu02%2Fitems%2F09%3Ff%3Djsonld%22%7D%2C%7B%22rel%22%3A%22alternate%22%2C%22type%22%3A%22text%2Fhtml%22%2C%22title%22%3A%22This%20document%20as%20HTML%22%2C%22href%22%3A%22https%3A%2F%2Finfo.geoconnex.us%2Fcollections%2Fhu02%2Fitems%2F09%3Ff%3Dhtml%22%7D%2C%7B%22rel%22%3A%22collection%22%2C%22type%22%3A%22application%2Fjson%22%2C%22title%22%3A%22HU02%22%2C%22href%22%3A%22https%3A%2F%2Finfo.geoconnex.us%2Fcollections%2Fhu02%22%7D%2C%7B%22rel%22%3A%22prev%22%2C%22type%22%3A%22application%2Fgeo%2Bjson%22%2C%22href%22%3A%22https%3A%2F%2Finfo.geoconnex.us%2Fcollections%2Fhu02%2Fitems%2F09%22%7D%2C%7B%22rel%22%3A%22next%22%2C%22type%22%3A%22application%2Fgeo%2Bjson%22%2C%22href%22%3A%22https%3A%2F%2Finfo.geoconnex.us%2Fcollections%2Fhu02%2Fitems%2F09%22%7D%5D%7D&frame=%7B%7D&context=%7B%7D))

I think using a blank node and `schema:url` to link to the geojson version of the feature is a non-unreasonable way to atleast get the geojson geometry into the linked data graph. 

The other solution would be to direct-encode the geometry as WKT but that's probably overkill and not any more useful than a link to the bare geojson.

```jsonld
{
  "@context": [
    "https://geojson.org/geojson-ld/geojson-context.jsonld",
    {
      "schema": "https://schema.org",
      "NAME": "schema:name",
      "GNIS_ID": "schema:subjectOf",
      "uri": "@id"
    }
  ],
  "type": "Feature",
  "properties": {
    "uri": "https://geoconnex.us/ref/hu02/09",
    "NAME": "Souris-Red-Rainy Region",
    "gnis_url": "https://geonames.usgs.gov/apex/f?p=gnispq:3:::NO::P3_FID:2730139",
    "GNIS_ID": 2730139,
    "HUC2": "09",
    "LOADDATE": "2019/05/23 18:26:19+00"
  },
  "geometry": {
    "schema:encodingFormat": "application/geo+json",
    "schema:url": "https://info.geoconnex.us/collections/hu02/items/09?f=json"
  },
  "id": "https://geoconnex.us/ref/hu02/09",
  "links": [
    {
      "rel": "alternate",
      "type": "application/geo+json",
      "title": "This document as GeoJSON",
      "href": "https://info.geoconnex.us/collections/hu02/items/09?f=json"
    },
    {
      "rel": "self",
      "type": "application/ld+json",
      "title": "This document as RDF (JSON-LD)",
      "href": "https://info.geoconnex.us/collections/hu02/items/09?f=jsonld"
    },
    {
      "rel": "alternate",
      "type": "text/html",
      "title": "This document as HTML",
      "href": "https://info.geoconnex.us/collections/hu02/items/09?f=html"
    },
    {
      "rel": "collection",
      "type": "application/json",
      "title": "HU02",
      "href": "https://info.geoconnex.us/collections/hu02"
    },
    {
      "rel": "prev",
      "type": "application/geo+json",
      "href": "https://info.geoconnex.us/collections/hu02/items/09"
    },
    {
      "rel": "next",
      "type": "application/geo+json",
      "href": "https://info.geoconnex.us/collections/hu02/items/09"
    }
  ]
}
```

**Comment by alpha-beta-soup** [comment]
That's definitely superior to my suggestion of making it `null`.

**Comment by tomkralidis** [comment]
Implemented in #565
